uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -47,14 +47,37 @@ define([
         },
 
         /**
-         * This method is used to update a leave request
+         * Update a leave request
          *
-         * @param {object} attributes - Values which needs to be updated
          * @return {Promise} Resolved with {Object} Updated Leave request
          */
-        update: function (attributes) {
-          return LeaveRequestAPI.sendPOST('LeaveRequest', 'create', _.assign({}, this.attributes(), attributes))
+        update: function () {
+          return LeaveRequestAPI.update(this.attributes());
+        },
+
+        /**
+         * Create a new leave request
+         *
+         * @return {Promise} Resolved with {Object} Created Leave request with
+         *  newly created id for this instance
+         */
+        create: function (attributes) {
+          return LeaveRequestAPI.create(this.attributes())
+            .then(function(result){
+              this.attributes = _.assign({}, this.attributes(), result);
```

Comment:
Why are you not simply doing `this.id = result.id` (or similar)? If you replace all the current properties of the instance with the object returned by the API, then that means that you would have to send that object through the `fromAPI` filter. I'm not sure it's worth to do it